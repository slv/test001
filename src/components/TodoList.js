import React from "react";
import PropTypes from "prop-types";
import { TodoList, TodoLoading } from "./TodoList.theme";

export const Component = ({ todo_list, todo_loading }) => (
  <TodoList>
    {todo_loading ? (
      <TodoLoading />
    ) : (
      (todo_list || []).map(todo_item => (
        <div key={todo_item.id}>{todo_item.title}</div>
      ))
    )}
  </TodoList>
);

Component.displayName = "TodoList";

Component.propTypes = {
  todo_list: PropTypes.array,
  todo_loading: PropTypes.bool
};

export default Component;
