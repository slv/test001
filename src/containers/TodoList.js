import React from "react";
import { connect } from "react-redux";
import { createSelector } from "reselect";
import { getTodoList, getTodoLoading } from "../selectors/getTodo";
import TodoList from "../components/TodoList";
import { apiGet } from "../actions/api";

class TodoListContainer extends React.Component {
  componentDidMount() {
    this.props.apiGet({
      path: "/todos",
      data: {
        _limit: 20
      }
    });
  }

  render() {
    return <TodoList {...this.props} />;
  }
}

const mapStateToProps = () =>
  createSelector(
    getTodoList,
    getTodoLoading,
    (todo_list, todo_loading) => {
      return {
        todo_list,
        todo_loading
      };
    }
  );

const mapDispatchToProps = {
  apiGet
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TodoListContainer);
