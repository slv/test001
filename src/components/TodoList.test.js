import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import TodoList from "./TodoList";

describe("TodoList", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TodoList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("match snapshot", () => {
    const renderedValue = renderer.create(<TodoList />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
