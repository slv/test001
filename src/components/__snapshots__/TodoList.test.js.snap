// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList match snapshot 1`] = `
<div
  className="sc-bdVaJa foRpMy"
/>
`;
